src/MatrixDotNet/Vector.cs

```diff
@@ -104,17 +104,27 @@ private static Vector<T> Sub(T[] a, T[] b)
             return c;
         }
 
-        private static T ScalarProduct(T[] a,T[] b)
+        private static T DotProduct(T[] a,T[] b)
```

Comment:
I can't find unit tests for vectors product, can you add them, please (or leave a link in comments if I'm not attentive enough 👀)